### PR TITLE
Geant simulation

### PR DESCRIPTION
**Pull request by zaborowska**
Geant simulation:
1. Detector TEMPORARY read from gdml file (FCCDetectorConstruction, FCCFullDetector.gdml to be changed/deleted) - to be replaced with the DD4hep detector
2. Physics list includes Fast SImulation Models
3. Event read in Gaudi from HepMC event-by-event and passed to Geant Run Manager
4. Output TEMPORARY saved with Geant tools (FCCOutput, FCCPrimaryParticleInformation, to be changed/deleted) - to be replaced with the EDM

Current stage does not include Atlfast: just simple parametrisation, pT-dependent.

### CODE REVIEW

File: GeantFast/src/FCCSmearer.cc [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+#include "G4UnitsTable.hh"
+#include "G4SystemOfUnits.hh"
+#include "G4TransportationManager.hh"
+#include "G4FieldManager.hh"
+#include "G4UniformMagField.hh"
+#include <ctime>
+
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
+FCCSmearer* FCCSmearer::fFCCSmearer = 0;
```
Comment (sbinet) [inline GeantFast/src/FCCSmearer.cc]: this should read nullptr_t

File: GeantFast/src/FCCSmearer.cc [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+   fRandomGauss = new CLHEP::RandGauss(fRandomEngine);
+}
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
+FCCSmearer::~FCCSmearer()
+{}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
+FCCSmearer* FCCSmearer::Instance()
+{
+   if(!fFCCSmearer)
```
Comment (sbinet) [inline GeantFast/src/FCCSmearer.cc]: that's racy and a one-time leak which pollutes valgrind reports. prefer this (still non-thread-safe):

```c++
static FCCSmearer smearer;
fFCCSmearer = &smearer;
return fFCCSmearer;
```

File: GeantFast/GeantFast/FCCSmearer.hh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+   A default constructor.
+*/
+   FCCSmearer();
+   ~FCCSmearer();
+
+private:
+/**
+   A pointer to FCCSmearer object.
+*/
+   static FCCSmearer* fFCCSmearer;
```
Comment (sbinet) [inline GeantFast/GeantFast/FCCSmearer.hh]: this doesn't look like a good idea, in a multithreaded world a la GaudiHive: a singleton with a PRNG as data member with no mutex lock won't fly in MT.

File: GeantFast/src/FCCSmearer.cc (hunk 1 above)
Comment (hegner) [inline GeantFast/src/FCCSmearer.cc]: actually nullptr ;-)

File: GeantFast/GeantFast/FCCSmearer.hh (hunk 3 above)
Comment (hegner) [inline GeantFast/GeantFast/FCCSmearer.hh]: correct. The entire G4 interface ATM is rather shaky. It's work in progress though.

Comment (sbinet) [inline GeantFast/GeantFast/FCCSmearer.hh]: of course, I was just being "cap'tain obvious" here (and not dismissive of anything in that PR), playing by the PR rules.

**Pull request merged**